pkg/controller/testutil/test_utils.go

```diff
@@ -417,6 +422,26 @@ func (f *FakeRecorder) AnnotatedEventf(obj runtime.Object, annotations map[strin
 	f.Eventf(obj, eventtype, reason, messageFmt, args...)
 }
 
+func (f *FakeRecorder) AssertEventReasons(events []v1.Event) bool {
+	f.Lock()
+	defer f.Unlock()
+	return f.assertReasons(events, f.Events)
+}
+
+func (f *FakeRecorder) assertReasons(test []v1.Event, expect []*v1.Event) bool {
```

Comment:
nit: usually assert functions call these parameters `actual` and `expected` and they're used in an opposite order, e.g. https://github.com/stretchr/testify/blob/master/assert/assertions.go#L156

Reply:
removed function but new call to slices.EqualFunc has correct order :)